tests/src/main/java/net/neoforged/neoforge/debug/transfer/VanillaHandlersTests.java

```diff
@@ -0,0 +1,566 @@
+
+        try (var tx = Transaction.open(null)) {
+            if (resourceHandler.insert(ItemResource.of(Items.SHULKER_BOX), 1, tx) > 0) {
+                helper.fail(Component.literal("Expected shulker box to be rejected"), pos);
+            }
+        }
+
+        helper.succeed();
+    }
+
+    /**
+     * {@link Container#canPlaceItem(int, ItemStack)} is supposed to be independent of the stack size.
+     * However, to limit some stackable inputs to a size of 1, brewing stands and furnaces don't follow this rule in all cases.
+     * This test ensures that the Transfer API works around this issue for furnaces.
+     */
+    @GameTest
+    @EmptyTemplate
+    @TestHolder
+    public static void testBadFurnaceIsValid(ExtendedGameTestHelper helper) {
+        BlockPos pos = new BlockPos(0, 1, 0);
+        helper.setBlock(pos, Blocks.FURNACE.defaultBlockState());
+        FurnaceBlockEntity furnace = helper.getBlockEntity(pos, FurnaceBlockEntity.class);
+        var furnaceWrapper = VanillaContainerWrapper.of(furnace);
+
+        try (Transaction tx = Transaction.open(null)) {
+            if (furnaceWrapper.insert(1, ItemResource.of(Items.BUCKET), 2, tx) != 1) {
+                throw helper.assertionException("Exactly 1 bucket should have been inserted");
+            }
+        }
+
+        helper.succeed();
+    }
+
+    /**
+     * Same as {@link #testBadFurnaceIsValid}, but for brewing stands.
+     */
+    @GameTest
+    @EmptyTemplate
+    @TestHolder
+    public static void testBadBrewingStandIsValid(ExtendedGameTestHelper helper) {
+        BlockPos pos = new BlockPos(0, 1, 0);
+        helper.setBlock(pos, Blocks.BREWING_STAND.defaultBlockState());
+        BrewingStandBlockEntity brewingStand = helper.getBlockEntity(pos, BrewingStandBlockEntity.class);
+        var brewingStandWrapper = VanillaContainerWrapper.of(brewingStand);
+
+        try (Transaction tx = Transaction.open(null)) {
+            for (int bottleSlot = 0; bottleSlot < 3; ++bottleSlot) {
+                if (brewingStandWrapper.insert(bottleSlot, ItemResource.of(Items.GLASS_BOTTLE), 2, tx) != 1) {
+                    throw helper.assertionException("Exactly 1 glass bottle should have been inserted");
+                }
+            }
+
+            if (brewingStandWrapper.insert(3, ItemResource.of(Items.REDSTONE), 2, tx) != 2) {
+                throw helper.assertionException("Brewing ingredient insertion should not be limited");
+            }
+        }
+
+        helper.succeed();
+    }
+
+    // TODO: I would like to bring this test over, but we need snbt test structure support first
+//    /**
+//     * Regression test for <a href="https://github.com/FabricMC/fabric/issues/2810">double chest wrapper only updating modified halves</a>.
+//     */
+//    @GameTest(structure = "fabric-transfer-api-v1-testmod:double_chest_comparators", skyAccess = true)
+//    public static void testDoubleChestComparator(TestContext context) {
```

Comment:
We should make sure this compiles and has the correct names (instead of being Yarn names), and just have the annotation be what is commented out with the TODO